baseball/src/App.js

```diff
@@ -0,0 +1,100 @@
+
+//IMPORTS ⬇︎
+import React, { Component } from 'react';
+import Score from './components/score';
+import Megatron from './components/megatron';
+import "./App.css";
+
+//APP SETUP  ⬇︎
+class App extends Component {
+  constructor( props ) {
+    super( props );
+    this.state = {
+      balls: 0,
+      strikes: 0
+    }
+  }
+
+  hit = e => {
+    this.setState({
+      balls: 0,
+      strikes: 0,
+    })
+  };
+
+  addBall = e => {
+    this.setState( prevState => {
+      return {
+        balls: prevState.balls + 1
+      }
+    })
+  };
+
+  addStrike = e => {
+    this.setState( prevState => {
+      return {
+        strikes: prevState.strikes + 1,
+      }
+    })
+  };
+
+  addFoul = e => {
+    if ( this.state.strikes === 0 ) {
+      this.setState( prevState => {
+        return {
+          strikes: prevState.strikes + 1
+        }
+      })
+    }
+    if ( this.state.strikes === 1 ) {
+      this.setState( prevState => {
+        return {
+          strikes: prevState.strikes + 1
+        }
+      })
+    }
+    if ( this.state.strikes === 2 ) {
+      this.setState( prevState => {
+        return {
+          strikes: prevState = 2
+        }
+      })
+    }
+  }
+
+  componentDidUpdate = ( prevProps , prevState ) => {
```

Comment:
great job using the CDU lifecycle for this. you could have also tested this in the hit and strike methods, but this is a much more functional programming paradigm approach.